src/OrbitGl/App.cpp

```diff
@@ -2649,6 +2649,17 @@ void OrbitApp::set_selected_thread_id(ThreadID thread_id) {
   return data_manager_->set_selected_thread_id(thread_id);
 }
 
+std::optional<orbit_client_data::ThreadStateSliceInfo> OrbitApp::selected_thread_state_slice()
+    const {
+  return data_manager_->selected_thread_state_slice();
+}
+
+void OrbitApp::set_selected_thread_state_slice(
+    std::optional<orbit_client_data::ThreadStateSliceInfo> thread_state_slice) {
+  RequestUpdatePrimitives();
+  return data_manager_->set_selected_thread_state_slice(thread_state_slice);
```

Comment:
1. Why `return`?
2. It would make more sense to me to call the setter first and then request the update.

Reply:
I removed the return but for point 2, the function `set_selected_thread_id` first requests the update then it updates the value so I followed that logic because it seems that's the way we're expected to use `RequestUpdatePrimitives`